weaviate/backup/backup.py

```diff
@@ -357,6 +365,75 @@ async def get_restore_status(
         """
         return await self.__get_restore_status(backup_id, backend)
 
+    async def __cancel_backup(self, backup_id: str, backend: BackupStorage) -> bool:
+        backup_id, backend = _get_and_validate_get_status(
+            backup_id=backup_id,
+            backend=backend,
+        )
+        path = f"/backups/{backend.value}/{backup_id}"
+
+        response = await self._connection.delete(
+            path=path, error_msg="Backup delete failed due to connection error."
+        )
+        typed_response = _decode_json_response_dict(response, "Backup restore status check")
+        if typed_response is None:
+            raise EmptyResponseException()
+
+        if response.status_code == 204:
+            return True  # Successfully deleted
+        else:
+            return False  # did not exist
+
+    async def cancel_backup(self, backup_id: str, backend: BackupStorage) -> bool:
+        """
+        Cancels a running backup.
+
+        Parameters
+        ----------
+        backup_id : str
+            The identifier name of the backup.
+            NOTE: Case insensitive.
+        backend : BackupStorage
+            The backend storage where to create the backup.
+
+        Raises
+        ------
+        weaviate.UnexpectedStatusCodeException
+            If weaviate reports a none OK status.
+
+        Returns
+        -------
+         A bool indicating if the cancellation was successful.
+        """
+        return await self.__cancel_backup(backup_id, backend)
+
+    async def __list_backups(self, backend: BackupStorage) -> List[BackupReturn]:
+        _, backend = _get_and_validate_get_status(backend=backend, backup_id="dummy")
+        path = f"/backups/{backend.value}"
+
+        response = await self._connection.get(
+            path=path, error_msg="Backup list status failed due to connection error."
+        )
+        typed_response = _decode_json_response_list(response, "Backup restore status check")
+        if typed_response is None:
+            raise EmptyResponseException()
+        return [BackupReturn(**entry) for entry in typed_response]
+
+    async def list_backups(self, backend: BackupStorage) -> List[BackupReturn]:
```

Comment:
This API call needs to be removed as it was disabled in the server side https://github.com/weaviate/weaviate/pull/5797